### PR TITLE
SONARXML-119 S2068 should not raise on android:password

### PR DESCRIPTION
**Pull request by dorian-burihabwa-sonarsource**

### CODE REVIEW

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java [hunk 1]
```diff
@@ -118,7 +118,15 @@ private void checkAttributes(Node node, Set<String> credentialWords, boolean rep
...
-    return localName != null && credentialWords.contains(localName.toLowerCase(Locale.ROOT));
+    if (localName == null || !credentialWords.contains(localName.toLowerCase(Locale.ROOT))) {
+      return false;
+    }
+    return !isTruthyAndroidPasswordAttribute(node);
+  }
+
+  private static boolean isTruthyAndroidPasswordAttribute(Node node) {
+    return node.getNodeName().equalsIgnoreCase("android:password") &&
+      node.getNodeValue().trim().equalsIgnoreCase("true");
```
Comment (quentin-jaquier-sonarsource) [inline sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java:129]: The rule aims to detect hardcoded credentials, this is not the case for `android:password` regardless of the value, we can probably remove this condition.

File: sonar-xml-plugin/src/test/resources/checks/HardcodedCredentialsCheck/android_password.xml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<RelativeLayout xmlns:android="http://schemas.android.com/apk/res/android" xmlns:app="http://schemas.android.com/apk/res-auto" android:layout_width="match_parent" android:layout_height="match_parent">
+  <ImageButton android:id="@+id/assistance_button" android:background="@null" android:layout_width="wrap_content" android:layout_height="wrap_content" android:layout_margin="@dimen/default_margin" android:src="@drawable/ic_actionbar_help" android:layout_alignParentRight="true"/>
+  <com.aaa.views.FontEditText android:id="@+id/edit_password_0" android:hint="@string/hint_password" android:password="true"  app:font="gotham-book-clean.ttf" style="@style/LoginEditText"/>
+  <com.aaa.views.FontEditText android:id="@+id/edit_password_1" android:hint="@string/hint_password" android:password="false" app:font="gotham-book-clean.ttf" style="@style/LoginEditText"/> <!-- Noncompliant {{"password" detected here, make sure this is not a hard-coded credential.}} -->
+</RelativeLayout>
```
Comment (quentin-jaquier-sonarsource) [inline sonar-xml-plugin/src/test/resources/checks/HardcodedCredentialsCheck/android_password.xml:6]: There is a lot of noise in this test file, it is hard to understand what we are testing. We can probably reduce it to keep less code, even if the code would not make sense in an app.

**Review by quentin-jaquier-sonarsource — CHANGES_REQUESTED** [review]

File: sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java (hunk 1 above)
Comment (dorian-burihabwa-sonarsource) [reply sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/checks/security/HardcodedCredentialsCheck.java:129]: Indeed, the ticket has been modified to reflect that

**Review by quentin-jaquier-sonarsource — APPROVED** [review]
LGTM

**Pull request merged**